admin-tools/data_mover.sh

```diff
@@ -0,0 +1,116 @@
+
+		if test $length -eq 0; then
+			echo "$srcClusterURL/webhdfs/v1$path/$entry is a directory"
+			# Create the directory in the destination cluster with appropriate ownership
+			curl -X PUT "$dstClusterURL/webhdfs/v1$path/$entry?op=mkdirs&user.name=hdfs" 2> /dev/null > /dev/null
+			curl -X PUT "$dstClusterURL/webhdfs/v1$path/$entry?op=setowner&owner=$username&group=$username&user.name=hdfs" \
+				2> /dev/null > /dev/null
+			# This directory must be migrated as well in a recursive fashion
+			migrate $path/$entry $username
+		else
+			echo "$srcClusterURL/webhdfs/v1$path/$entry is a file ($length bytes)"
+			# Create a temporal file for downloading the entire HDFS file
+			tmpFile=$(mktemp $tmpDir/file.XXXX)
+			# Download the HDFS file
+			curl -X GET "$srcClusterURL/webhdfs/v1$path/$entry?op=open&user.name=hdfs" > $tmpFile 2> /dev/null
+			# Get the number of chunks according to the maximum number of bytes allowed for transferring
+			numChunks=$(($length/$maxBytesTransfer))
+
+			# Copy each chunk into a HDFS file in the destination cluster
+			for i in `seq 0 $numChunks`; do
+				# Get the offset and number of bytes to be transferred for the current chunk
+				offset=$(($i*$maxBytesTransfer))
+
+				if test $i -eq $numChunks; then
+					bytes=$(($length-$offset))
+				else
+					bytes=$maxBytesTransfer
+				fi
+
+				# Depending on the chunk, the file must be created with initial content or the data appended
+				if test $i -eq 0; then
+					# This is the first chunk, thus create the file
+					dd if=$tmpFile ibs=1 skip=0 count=$bytes 2> /dev/null | \
+						curl -L --data-binary @- -X PUT "$dstClusterURL/webhdfs/v1$path/$entry?op=create&user.name=hdfs" \
+						2> /dev/null > /dev/null
+				else
+					# This is other chunk, thus append the data
+					dd if=$tmpFile ibs=1 skip=$offset count=$bytes 2> /dev/null | \
+						curl -L --data-binary @- -X POST "$dstClusterURL/webhdfs/v1$path/$entry?op=append&user.name=hdfs" \
+						2> /dev/null > /dev/null
+				fi
+			done
+
+			# Change the ownership of the new HDFS file
+			curl -X PUT "$dstClusterURL/webhdfs/v1$path/$entry?op=setowner&owner=$username&group=$username&user.name=hdfs" \
+				2> /dev/null > /dev/null
+			# Remove the temporal file
+			rm $tmpFile
+		fi
+
+		index=$((index+1))
+	done < <(curl -X GET "$srcClusterURL/webhdfs/v1$path?op=liststatus&user.name=hdfs" 2> /dev/null | \
+		python -m json.tool | grep pathSuffix | awk '{ print $2 }' | sed 's/"//g' | sed 's/,//g')
+}
+
+# Main function
+while read -r username; do
+	# Create the base path in the destination cluster
+	curl -X PUT "$dstClusterURL/webhdfs/v1/user/$username?op=mkdirs&user.name=hdfs" 2> /dev/null > /dev/null
+	curl -X PUT "$dstClusterURL/webhdfs/v1/user/$username?op=setowner&owner=$username&group=$username&user.name=hdfs" \
+		2> /dev/null > /dev/null
+	# Start by migrating the base path
+	migrate /user/$username $username
+done < <(cat $usernamesFile)
+
+# Remove the temporal working directory
```

Comment:
I understand that a temporal directory in the local machine where the script is running is used to do the migration. Do you check if the available disk space is enough (it could be a problem in the case of large data sets)?